### PR TITLE
Update "project" to mention pyproject.toml, not just setuptools

### PR DESCRIPTION
**Pull request by encukou**
Also, add the pyproject.toml and setup.cfg markers as other possible "marker" files for a "practical way to define projects".

---

The [project](https://packaging.python.org/glossary/#term-project) entry is currently outdated. Here's my suggestion for updating it.
(Is a PR is the right way to suggest the change, or should I discuss on Discourse first?)

### CODE REVIEW

File: source/glossary.rst [hunk 1]
```diff
@@ -102,11 +102,10 @@ Glossary
         packaged into a :term:`Distribution <Distribution Package>`.
 
         Since most projects create :term:`Distributions <Distribution Package>`
-        using :ref:`distutils` or :ref:`setuptools`, another practical way to
-        define projects currently is something that contains a :term:`setup.py`
-        at the root of the project src directory, where "setup.py" is the
-        project specification filename used by :ref:`distutils` and
-        :ref:`setuptools`.
+        using either :pep:`518` ``build-system``, :ref:`distutils` or
+        :ref:`setuptools`, another practical way to define projects currently
```
Comment (webknjaz) [inline source/glossary.rst:106]: I'd also swap distutils/setuptools to keep have list in the order of priority:
```suggestion
        using either :pep:`518` ``build-system``, :ref:`setuptools` or
        :ref:`distutils`, another practical way to define projects currently
```

**Comment by encukou** [comment]
I'm not sure who should do the final approval/merge.
@webknjaz's change looks good; if you agree, please commit/squash/apply it as necessary when merging.

File: source/glossary.rst [hunk 2]
```diff
@@ -102,11 +102,10 @@ Glossary
...
+        is something that contains a :term:`pyproject.toml`, :term:`setup.py`,
+        or :term:`setup.cfg` file at the root of the project source directory.
```
Comment (pfmoore) [inline source/glossary.rst:108]: Is it valid to have `setup.cfg` without `setup.py`? That seems like an unrelated change, even if it is true. Maybe we should stick to just adding `pyproject.toml`?

Comment (webknjaz) [reply source/glossary.rst:108]: @pfmoore it is valid to have `setup.cfg` w/o `setup.py` if you have a modern enough setuptools version. Like in case with `setuptools` mentioned as a build dep in `pyproject.toml`.

**Comment by webknjaz** [comment]
@encukou I don't have merge powers in this repo so it's not up to me.

File: source/glossary.rst (hunk 2 above)
Comment (pfmoore) [reply source/glossary.rst:108]: OK, but in that case you need a `pyproject.toml`, so it seems a bit redundant. No big deal, though.
